contracts/integrations/custom/CustomIntegrationBalancerv2.sol

```diff
@@ -0,0 +1,416 @@
+     * @return uint256                   Call value
+     * @return bytes                     Trade calldata
+     */
+    function _getExitCalldata(
+        address _strategy,
+        bytes calldata _data,
+        uint256 _resultTokensIn,
+        address[] calldata _tokensOut,
+        uint256[] calldata _minAmountsOut
+    )
+        internal
+        view
+        override
+        returns (
+            address,
+            uint256,
+            bytes memory
+        )
+    {
+        bytes32 poolId = BasePool(BytesLib.decodeOpDataAddress(_data)).getPoolId();
+        address strategy = _strategy;
+        uint256 resultTokensIn = _resultTokensIn;
+
+        (IERC20[] memory tokens, , ) = IVault(vaultAddress).getPoolTokens(poolId);
+        require(_tokensOut.length == tokens.length, 'Must supply same number of tokens as are already in the pool!');
+
+        IVault.ExitPoolRequest memory exitRequest = _getExitRequest(_tokensOut, tokens, _minAmountsOut, resultTokensIn);
+        bytes memory methodData = abi.encodeWithSelector(
+            IVault.exitPool.selector,
+            poolId,
+            strategy,
+            strategy,
+            exitRequest
+        );
+
+        return (vaultAddress, 0, methodData);
+    }
+
+    /**
+     * The list of addresses of the IERC-20 tokens mined as rewards during the strategy
+     *
+     * hparam  _data                      Address provided as param
+     * @return address[] memory           List of reward token addresses
+     */
+    function _getRewardTokens(
+        address /* _data */
+    ) internal pure override returns (address[] memory) {
+        // No extra rewards.
+        return new address[](1);
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * The tokens to be purchased by the strategy on enter according to the weights.
+     * Weights must add up to 1e18 (100%)
+     *
+     * hparam  _data                      Address provided as param
+     * @return _inputTokens               List of input tokens to buy
+     * @return _inputWeights              List of weights for the tokens to buy
+     */
+    function getInputTokensAndWeights(bytes calldata _data)
+        external
+        view
+        override
+        returns (address[] memory _inputTokens, uint256[] memory _inputWeights)
+    {
+        BasePool pool = BasePool(BytesLib.decodeOpDataAddressAssembly(_data, 12));
+        IVault vault = IVault(vaultAddress);
+        bytes32 poolId = pool.getPoolId();
+
+        (IERC20[] memory poolTokens, uint256[] memory balances, ) = vault.getPoolTokens(poolId);
+
+        uint256 tokenBalanceTotal;
+        _inputWeights = new uint256[](poolTokens.length);
+        _inputTokens = new address[](poolTokens.length);
+
+        for (uint8 i = 0; i < poolTokens.length; ++i) {
+            tokenBalanceTotal += _getBalanceFullDecimals(balances[i], poolTokens[i]).mul(
+                _getPrice(address(poolTokens[i]), USDC)
```

Comment:
Note that we convert all assets in a pool into USDC to find out the correct value distribution when entering the pool. The choice to use USDC was made because we assumed that the Babylon oracle should have a price for most assets in USDC. Is there some other currency we should be using instead?

Reply:
USDC, WETH or DAI are the best. Maybe WETH is slightly better

Reply:
Okay, changing to WETH.